zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/bpmn/container/CallActivityProcessor.java

```diff
@@ -206,15 +207,16 @@ private Either<Failure, DirectBuffer> evaluateProcessId(
         processIdExpression, scopeKey);
   }
 
-  private Either<Failure, DeployedProcess> getProcessForProcessIdAndBindingType(
+  private Either<Failure, DeployedProcess> getCalledProcess(
       final DirectBuffer processId,
       final ZeebeBindingType bindingType,
+      final String versionTag,
       final BpmnElementContext context) {
     return switch (bindingType) {
       case deployment -> getProcessVersionInSameDeployment(processId, context);
       case latest -> getLatestProcessVersion(processId, context.getTenantId());
-      // will be implemented with https://github.com/camunda/camunda/issues/21038
-      case versionTag -> Either.left(new Failure("Binding type 'versionTag' not supported"));
+      case versionTag ->
+          getLatestProcessVersionWithVersionTag(processId, versionTag, context.getTenantId());
```

Comment:
❓ Should we have a `default` case that throws an error since users may manually modify a BPMN XML and provide a faulty value for the binding type?

Reply:
Actually, that case should already be covered by the `ZeebeBindingTypeValidator`, so users would not be able to deploy a model with an invalid binding type in the first place 🙂.